cloud_functions/function.js

```diff
@@ -0,0 +1,8 @@
+const flagCryptominer = require('./flagCryptominer')
+const { getMixpanel } = require('./mixpanel')
+
+
+module.exports.flagCryptominer = async (message, context) => {
+  const mixpanel = await getMixpanel()
+  flagCryptominer.run(mixpanel, message)
```

Comment:
Assuming this kicks off an async call, we should make sure we're waiting for it to finish before returning. Otherwise it's possible the instance could be shut down with unfinished requests in rare cases. (That might mean promisifying the mixpanel call further down if needed)

Reply:
I'm pretty sure I dug into this and found that it's not making an async call. o_O I'll dig again because it is very strange, and maybe worth a comment since it does look wrong.

Reply:
I glanced at the docs and they're not super clear, but based on the current signature of `people.set`, it looks like it might be callback based. If that's the case, then we'd need to wrap the call in an explicit Promise to wire everything up.

Reply:
Ah, yes. Thanks for catching this! I so remember reading about callback-based APIs, but I guess I came to the wrong conclusion about what I needed to do. I'll fix this.

Reply:
After going through the exercise of writing it myself (which was a fun little exercise), I ended up using Node's `promisify` util function. Combined with the `withErrorReporting` suggestion, `flagCryptominer.js` now looks quite tidy! Thanks again for all the suggestions.